aws/resource_aws_lex_slot_type_test.go

```diff
@@ -0,0 +1,146 @@
+					resource.TestCheckResourceAttr(resourceName, "enumeration_value.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "name", testSlotTypeID),
+					resource.TestCheckResourceAttr(resourceName, "value_selection_strategy", lexmodelbuildingservice.SlotValueSelectionStrategyOriginalValue),
+
+					// computed attributes
+					resource.TestCheckResourceAttrSet(resourceName, "checksum"),
+					resource.TestCheckResourceAttrSet(resourceName, "version"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: fmt.Sprintf(testAccAwsLexSlotTypeUpdateConfig, testID),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// user defined attributes
+					resource.TestCheckResourceAttr(resourceName, "description", "Allowed flower types"),
+					resource.TestCheckResourceAttr(resourceName, "enumeration_value.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "name", testSlotTypeID),
+					resource.TestCheckResourceAttr(resourceName, "value_selection_strategy", lexmodelbuildingservice.SlotValueSelectionStrategyTopResolution),
+
+					// computed attributes
+					resource.TestCheckResourceAttrSet(resourceName, "checksum"),
+					resource.TestCheckResourceAttrSet(resourceName, "version"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsLexSlotTypeExists(slotTypeName, slotTypeVersion string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).lexmodelconn
+
+		_, err := conn.GetSlotType(&lexmodelbuildingservice.GetSlotTypeInput{
+			Name:    aws.String(slotTypeName),
+			Version: aws.String(slotTypeVersion),
+		})
+		if err != nil {
+			if isAWSErr(err, lexmodelbuildingservice.ErrCodeNotFoundException, "") {
+				return fmt.Errorf("error slot type %s not found, %s", slotTypeName, err)
+			}
+
+			return fmt.Errorf("error getting slot type %s: %s", slotTypeName, err)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckAwsLexSlotTypeDestroy(slotTypeName, slotTypeVersion string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).lexmodelconn
+
+		_, err := conn.GetSlotType(&lexmodelbuildingservice.GetSlotTypeInput{
+			Name:    aws.String(slotTypeName),
+			Version: aws.String(slotTypeVersion),
+		})
+
+		if err != nil {
+			if isAWSErr(err, lexmodelbuildingservice.ErrCodeNotFoundException, "") {
+				return nil
+			}
+
+			return fmt.Errorf("error getting slot type %s: %s", slotTypeName, err)
+		}
+
+		return fmt.Errorf("error slot type still exists after delete, %s", slotTypeName)
+	}
+}
+
+const testAccAwsLexSlotTypeConfig = `
+resource "aws_lex_slot_type" "test" {
+  description = "Types of flowers to pick up"
+
+  enumeration_value {
+    synonyms = [
+      "Lirium",
+    ]
+
+    value = "lilies"
+  }
+
+  name                     = "test_slot_type_%s"
```

Comment:
Since you are passing in TestId above which contains `test_slot_type_` you can simplify this to 
```suggestion
  name                     = %q
```